python/mxnet/gluon/block.py

```diff
@@ -253,7 +253,7 @@ def collect_params(self, select=None):
         The selected :py:class:`ParameterDict`
         """
         ret = ParameterDict(self._params.prefix)
-        if select is None:
+        if not select:
```

Comment:
if select = `[]` or `{}` or `''` or `()` then not select will be True, may be wrong here.

Reply:
You're right, and it should be fine as long as both empty string and None returns true.